File: C++/Level.2/H-Index.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> citations) {
    sort(citations.begin(), citations.end());
    int n = citations.size();
    int h = 0;
    for (int i = 0; i < n; ++i) {
        h = max(h, min(n - i, citations[i]));
    }
    return h;
}

File: JavaScript/Level.2/H-Index.js
function solution(citations) {
    var answer = 0;
    let n = citations.length;
    citations
        .sort((a, b) => a - b)
        .forEach((citation, index) => {
            answer = Math.max(answer, Math.min(citation, n - index));
        });
    return answer;
}
